### PR TITLE
Run ci on perlmutter and add method to run on perlmutter 

### LINKED ISSUE #1301: Run CI on Perlmutter
Action Items
- [x] Rename https://github.com/buildtesters/buildtest/blob/devel/.gitlab/cori.yml file to something like `nersc.yml` 
- [x] Rename https://github.com/buildtesters/buildtest/blob/devel/tests/test_cori.py to `test_nersc.py` and add method to run on Perlmutter
- [x] Make use of new gitlab runner to run jobs via user [bdtest](https://iris.nersc.gov/user/99914/info)

### PR DESCRIPTION
**Pull request by prathmesh4321**
Hi @shahzebsiddiqui . Can you please review the PR and suggest any required changes ?

### CODE REVIEW

File: tests/settings/nersc.yml [hunk 1]
```diff
@@ -56,3 +56,23 @@ system:
...
+
+  perlmutter:
+    hostnames: ["login*"]
+    moduletool: lmod
+    compilers:
+      enable_prgenv: true
+      prgenv_modules:
+        gcc: PrgEnv-gnu
+        cray: PrgEnv-cray
+        nvhpc: PrgEnv-nvidia
```
Comment (shahzebsiddiqui) [inline tests/settings/nersc.yml]: lets change this to `PrgEnv-nvhpc` i think we will start using this going forward.

**Comment by shahzebsiddiqui** [comment]
> Hi @shahzebsiddiqui . Can you please review the PR and suggest any required changes ?

see CI pipeline https://software.nersc.gov/NERSC/buildtest/-/jobs/186955 it looks like it failed. You need to define an executor. Atleast for now you should have https://github.com/buildtesters/buildtest-nersc/blob/devel/config.yml#L458-L474.

You may want to define a `slurm` executor eventually and then start adding batch tests for slurm scheduler

File: tests/settings/nersc.yml (hunk 1 above)
Comment (prathmesh4321) [reply tests/settings/nersc.yml]: Done !

**Comment by prathmesh4321** [comment]
> > Hi @shahzebsiddiqui . Can you please review the PR and suggest any required changes ?
> 
> see CI pipeline https://software.nersc.gov/NERSC/buildtest/-/jobs/186955 it looks like it failed. You need to define an executor. Atleast for now you should have https://github.com/buildtesters/buildtest-nersc/blob/devel/config.yml#L458-L474.
> 
> You may want to define a `slurm` executor eventually and then start adding batch tests for slurm scheduler

@shahzebsiddiqui I have added executors and the pipeline ran without errors. Do you want me to also add the slurm executors in this PR ?

**Comment by shahzebsiddiqui** [comment]
> > > Hi @shahzebsiddiqui . Can you please review the PR and suggest any required changes ?
> > 
> > 
> > see CI pipeline https://software.nersc.gov/NERSC/buildtest/-/jobs/186955 it looks like it failed. You need to define an executor. Atleast for now you should have https://github.com/buildtesters/buildtest-nersc/blob/devel/config.yml#L458-L474.
> > You may want to define a `slurm` executor eventually and then start adding batch tests for slurm scheduler
> 
> @shahzebsiddiqui I have added executors and the pipeline ran without errors. Do you want me to also add the slurm executors in this PR ?

this looks good for now but yes we probably do need to add more tests in the near future basically to replicate what was done on cori since the system will go down.

So please do add similar test to the ones found in https://github.com/buildtesters/buildtest/tree/devel/tests/examples/cori.  You can ignore https://github.com/buildtesters/buildtest/blob/devel/tests/examples/cori/burstbuffer.yml test since we dont have burst buffer on Perlmutter but all other test are valid. Note the test will need to be adapted for Perlmutter.

**Review by shahzebsiddiqui — APPROVED** [review]
Looks good for initial implementation

**Pull request merged**